python/whylogs/core/utils/protobuf_utils.py

```diff
@@ -36,16 +39,45 @@ def read_delimited_protobuf(stream: IO[bytes], proto_class_name: Type[T], offset
         return proto_class_name()
     buf = stream.read(size)
     msg = proto_class_name()
-    try:
-        msg.ParseFromString(buf)
-    except DecodeError as decode_error:
-        logger.warning(
-            f"{decode_error}: when reading delimited protobuf file of size "
-            f"{size} bytes as {proto_class_name}, falling back to non-delimited read."
-        )
+    delimited_read_failure = False
+
+    with warnings.catch_warnings(record=True) as warning_messages:
+        try:
+            logger.info(f"Attempting to read profile file as delimited from position {offset}.")
+            msg.ParseFromString(buf)
+        except Exception as decode_error:
+            logger.warning(
+                f"{decode_error}: when reading delimited protobuf file of size "
+                f"{size} bytes as {proto_class_name}, falling back to non-delimited read."
+            )
+            delimited_read_failure = True
+
+        if not delimited_read_failure and len(warning_messages) > 0:
+            for w in warning_messages:
+                if issubclass(w.category, RuntimeWarning):
```

Comment:
RuntimeWarning looks pretty general.

Is it possible to reach a case where it's a delimited file and we parsed it correctly, but another type of RuntimeWarning was raised, and then we incorrectly assume we should fall back to not-delimited parsing?

Reply:
Good point, tightened this to the specific DecodeError message string we see in testing.